buildconfig/CMake/Sanitizers.cmake

```diff
@@ -0,0 +1,63 @@
+#################################################
```

Comment:
Could https://github.com/arsenm/sanitizers-cmake be used instead of reinventing the wheel?

Reply:
A really good point, I originally looked at this. However I decided against it as I have some concerns that the project is unmaintained. There is currently a PR approved sat there since 2019.

If upstream ever breaks we will either have to patch the submodule, or replace it with this. 

If it was more than 1 file I'd be inclined to use upstream, however for a single file of fairly basic ifs and the ability to maintain it as we need I'm inclined to stick with this?

Reply:
A really good point, I originally looked at this. However I decided against it as I have some concerns that the project is unmaintained. There is currently a PR approved sat there since 2019.

If upstream ever breaks we will either have to patch the submodule, or replace it with this. 

If it was more than 1 file I'd be inclined to use upstream, however for a single file of fairly basic ifs and the ability to maintain it as we need I'm inclined to stick with this?